omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/TaskActionResource1_8.java

```diff
@@ -0,0 +1,221 @@
+	
+	private TaskFactoryWrapper taskFactoryWrapper = new TaskFactoryWrapper();
+	
+	public void setTaskFactoryWrapper(TaskFactoryWrapper taskFactoryWrapper) {
+		this.taskFactoryWrapper = taskFactoryWrapper;
+	}
+	
+	@Override
+	public Object create(SimpleObject post, RequestContext context) throws ResponseException {
+		TaskAction action = newDelegate();
+		
+		setConvertedProperties(action, post, getCreatableProperties(), true);
+		
+		Collection<TaskDefinition> taskDefinitions;
+		if (action.isAllTasks() != null && action.isAllTasks()) {
+			taskDefinitions = taskFactoryWrapper.getRegisteredTasks();
+			action.setTasks(new ArrayList<TaskDefinition>(taskDefinitions));
+		} else {
+			taskDefinitions = action.getTasks();
+		}
+		
+		ServletContext servletContext = getServletContext(context);
+		
+		if (action.getAction().equals("SCHEDULETASK") || action.getAction().equals("RESCHEDULETASK")
+		        || action.getAction().equals("DELETE") || action.getAction().equals("SHUTDOWNTASK")) {
+			if (taskDefinitions == null || taskDefinitions.isEmpty()) {
+				throw new IllegalRequestException("Cannot execute action " + action.getAction()
+				        + " on empty set of task definitions.");
+			}
+		}
+		if (action.isAllTasks() == null || !action.isAllTasks()) {
+			for (TaskDefinition taskDef : taskDefinitions) {
+				if (taskDef == null) {
+					throw new IllegalRequestException(
+					        "One or more of the task definitions you specified are not registered on this server");
+				}
+			}
+		}
+		
+		switch (action.getAction()) {
+			case SCHEDULETASK:
+				scheduleTask(taskDefinitions);
+				break;
+			case RESCHEDULETASK:
+				reScheduleTask(taskDefinitions);
+				break;
+			case DELETE:
+				deleteTask(taskDefinitions);
+				break;
+			case SHUTDOWNTASK:
+				shutDownTask(taskDefinitions);
+				break;
+			case RESCHEDULEALLTASK:
+				reScheduleAllTask();
+				break;
+			case ONSTARTUP:
+				onStartUpTask();
+				break;
+			case ONSHUTDOWN:
+				onShutDownUpTask();
+				break;
+		}
+		return ConversionUtil.convertToRepresentation(action, Representation.DEFAULT);
+	}
+	
+	private void scheduleTask(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskFactoryWrapper.scheduleTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new UnsupportedOperationException();
+			}
+		}
+	}
+	
+	private void shutDownTask(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskFactoryWrapper.shutDownTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new UnsupportedOperationException();
+			}
+		}
+	}
+	
+	// Stop and start a scheduled task.
+	private void reScheduleTask(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskFactoryWrapper.reScheduleTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new UnsupportedOperationException();
```

Comment:
I see so many UnsupportedOperationException. Does this mean if anything does wrong during these actions, it means that the rest call is not supported?

Reply:
I couldn't throw the SchedulerException , because method implemented from Creatable Interface doesn't  allow to throw the SchedulerException. Else I wanted to change the Creatable Interface. So I decided to throw UnsupportedOperationException

Reply:
Shall I throw ResponseException instead of UnsupportedOperationException?

Reply:
Since you are not doing anything with the thrown exception, just do not catch it.

Reply:
If I do not catch it, then want to throw it by methods. 
If I throw it from here, then want to throw it from Create method.
But I can't throw it because It is a overridden method. So I caught and throw like this. 
Are there any perfect methods for this?

Reply:
@dkayiwa Shall I throw like this, 
throw new ResourceDoesNotSupportOperationException("SchedulerException");

Reply:
@dkayiwa  If I throw the exception from here, then I can catch this from the angular REST request to indicate the user as well

Is it good to use like,
**throw ObjectNotFoundException("SchedulerException : Ca not execute the action")**

Reply:
Do something along these lines:
throw new APIException("Errors occurred while rescheduling task", e);

Reply:
Changed! Thanks @dkayiwa  :-)